app/stores/MessageUnreadStore.js

```diff
@@ -90,8 +99,10 @@ let UnreadStore = assign({}, BaseStore, {
                     }
 
                     _unreadBool = _unreadBool.set(backEndChannelId, _unread[backEndChannelId].latestMessageId > _unread[backEndChannelId].lastSeenMessageId);
+                    getMessagesCount(backEndChannelId, {id:_unread[backEndChannelId].lastSeenMessageId?_unread[backEndChannelId].lastSeenMessageId:0});
```

Comment:
Since getMessagesCount is a async call, we cannot know how long it takes to get the result. Even though this problem will be solved if you move this function to "action" part, I'd like to make you understand the javascript programming style.